lib/console.js

```diff
@@ -245,7 +245,12 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
-module.exports = new Console(process.stdout, process.stderr);
+if (process.binding('worker').threadId === 0) {
+  module.exports = new Console(process.stdout, process.stderr);
+} else {
+  const { SyncWriteStream } = require('internal/fs');
+  module.exports = new Console(new SyncWriteStream(1), new SyncWriteStream(2));
```

Comment:
A problem I've seen previously in other multithreaded applications is that sometimes the console output from different threads gets multiplexed like

normal:
```
message 1
message 2
```

multiplexed:

```
message message 1
2
```

Are we certain that the same would not happen here?

Reply:
The main thread is an async stream, so likely messages from the main thread would print well after messages from worker threads anyway. I think this most likely *would* suffer from that same problem though, given that there's no synchronization of the logging.

I think for this to work fully correctly, logging from the main thread and all worker threads might need to be coordinated through a single, separate logging thread.

Reply:
@Qard right... either that, or just add a mutex/`Atomics.wait()` protecting any attempt to use the console.

Reply:
Ugh … yeah. I’m not sure what I’d prefer yet, or how much of a real problem this is (all of these streams *are* sync under most circumstances, so I’d expect this to be a non-issue most or all of the time…)